Packages/io.chainsafe.web3-unity.web3auth/Runtime/Web3AuthWalletExtensions.cs

```diff
@@ -34,9 +37,13 @@ public static IWeb3ServiceCollection UseWeb3AuthWallet(this IWeb3ServiceCollecti
         collection.AssertServiceNotBound<ISigner>();
         collection.AssertServiceNotBound<ITransactionExecutor>();
 
-        collection.AddSingleton<ISigner, ITransactionExecutor, ILifecycleParticipant, Web3AuthWallet>();
-
-        collection.AddSingleton<IWeb3AuthTransactionHandler, Web3AuthTransactionHandler>();
+        collection.AddSingleton<IWalletProvider, Web3AuthProvider>();
+        
+        collection.AddSingleton( new AccountProvider());
```

Comment:
I would call this something like InProcessAccountProvider to reflect the fact that it is "InProcess" system related.

Another idea, it looks like Signer manages the state of AccountProvider. We could potentially make Signer implement IAccountProvider and just add this contract name when registering InProcessSigner-s in DI. This way the Signer would keep the Account object and you won't need to add the public Initialize method to IAccountProvider, making it a bit more secure.

Reply:
that's kinda challenging because `InProcessSigner` would still need a private key and the whole point of the account provider is providing private key xD, however we can make `Web3AuthProvider` implement `IAccountProvider` it would decrease some LOC and make the solution lighter

Reply:
https://github.com/ChainSafe/web3.unity/pull/1051#discussion_r1670623731 for this one, this is an implementation I inherited from the [Web3Auth GUI](https://github.com/ChainSafe/web3.unity/pull/1019), you can kind of see that I only changed the service I'm getting and I'm not sure if I can simply make the Monobehaviour injectable I would kind of need to refactor the whole thing which is out of scope but definitely needs attention, we had to let the Web3Auth GUI PR in to meet deadlines.

Reply:
done

Reply:
InProcessSigner would still have access to the private key as it has now, but the Account reference will reside and be set in the IProcessSigner itself and not the separate service (AccountProvider), effectively removing need in the public Initialize method for IAccountProvider.

Reply:
Oh, I see. Web3AuthProvider sets the account and not the signer.